File: only code/1614/1614.cpp
#include <iostream>
using namespace std;

int main(){
    long finger_num,countable;
    cin >> finger_num >> countable;
    long count;
    if (finger_num != 1 && finger_num != 5){
        count = (countable / 2) * 8;
        if (countable %2 == 1){
            count = count + 9 -finger_num;
        }
        else{
            count = count + finger_num - 1;
        }
    }
    else{
        count = countable * 8 + finger_num - 1;
    }
    cout << count;
    return 0;
}

File: only code/1614/1614.py
finger_num = int(input())
countable = int(input())
count = 0
while countable !=0 :
    if (count//4)%2 == 0:
        if finger_num != 5:
            countable -= 1
        count += 4
    else :
        if finger_num != 1 :
            countable -= 1
        count += 4
if (count//4)%2 == 0:
    print(count + finger_num - 1)
else:
    print(count - finger_num + 5)